### PR TITLE
[13.0][ADD] rma_delivery: New module

### PR DESCRIPTION
**Pull request by chienandalu**
This module extends the functionality of the RMA module to allow to choose a default
behavior for the shipping method used on the RMA returns to the customer.

Includes `rma` tests declaration unplug from tests cases, so we can reuse them properly in third modules

cc @Tecnativa TT33975

please review @pedrobaeza @ernestotejeda

### CODE REVIEW

File: rma_delivery/README.rst [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+**Table of contents**
+
+.. contents::
+   :local:
+
+Configuration
+=============
+
+If you want RMAs to be created from incoming emails, you need to:
```
Comment (pedrobaeza) [inline rma_delivery/README.rst]: incoming emails?

File: rma_delivery/README.rst [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+#. Go to *Inventory > Configuration > Settings*
+#. Choose an *RMA delivery method strategy*.
+#. There are 3 possibilities:
+
+   - Fixed method: the method will always be the same.
```
Comment (pedrobaeza) [inline rma_delivery/README.rst]:
```suggestion
   - Fixed method: the method will always be the same. Select it on the field X (or leave it empty for no delivery method at all).
```

File: rma_delivery/README.rst [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+     the fixed one will be chosen.
+
+Usage
+=====
+
+To use this module, you need to:
+
+#. Go to a RMA which has received quantities.
+#. Return it to the customer.
+#. Depending on your company configuration the picking will get one or another carrier.
```
Comment (pedrobaeza) [inline rma_delivery/README.rst]:
```suggestion
#. Depending on your company configuration, the return picking will get one or another carrier.
```

File: rma_delivery/models/rma.py [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+        strategy = picking_form.company_id.rma_delivery_strategy
+        delivery_method = picking_form.company_id.rma_fixed_delivery_method
+        partner_method = (
+            picking_form.partner_id.property_delivery_carrier_id
+            or picking_form.partner_id.commercial_partner_id.property_delivery_carrier_id
+        )
+        if (
+            strategy == "customer_method"
+            or strategy == "mixed_method"
+            and partner_method
```
Comment (pedrobaeza) [inline rma_delivery/models/rma.py]: Make sure of the operator precedence

```suggestion
            or (strategy == "mixed_method" and partner_method)
```

File: rma_delivery/tests/test_rma_delivery.py [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo.tests import Form, SavepointCase
+
+
+class TestRma(SavepointCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.company = cls.env.user.company_id
```
Comment (pedrobaeza) [inline rma_delivery/tests/test_rma_delivery.py]: No possible to inherit from a RMA base test? Moment to do it?

File: rma_delivery/README.rst (hunk 1 above)
Comment (chienandalu) [reply rma_delivery/README.rst]: Copypasted... fixed

File: rma_delivery/README.rst (hunk 2 above)
Comment (chienandalu) [reply rma_delivery/README.rst]: done

File: rma_delivery/models/rma.py (hunk 4 above)
Comment (chienandalu) [reply rma_delivery/models/rma.py]: done

File: rma_delivery/README.rst (hunk 3 above)
Comment (chienandalu) [reply rma_delivery/README.rst]: done

File: rma_delivery/tests/test_rma_delivery.py (hunk 5 above)
Comment (chienandalu) [reply rma_delivery/tests/test_rma_delivery.py]: Ok, done :)

File: rma/tests/test_rma.py [hunk 6]
```diff
@@ -123,6 +123,12 @@ def _create_delivery(self):
         picking.button_validate()
         return picking
 
+
+class TestRmaCase(TestRma):
+    @classmethod
+    def setUpClass(cls):
```
Comment (pedrobaeza) [inline rma/tests/test_rma.py]: This method override is not needed

File: rma_delivery/models/rma.py [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import models
+
+
+class Rma(models.Model):
+    _inherit = "rma"
+
+    def _prepare_returning_picking(self, picking_form, origin=None):
+        """Gather the company option for default carrier on RMA returns. We could
+        either
```
Comment (pedrobaeza) [inline rma_delivery/models/rma.py]: This comment seems half-completed

Comment (chienandalu) [reply rma_delivery/models/rma.py]: Done

File: rma/tests/test_rma.py (hunk 6 above)
Comment (chienandalu) [reply rma/tests/test_rma.py]: Fixed

**Review by pedrobaeza — APPROVED** [review]

**Review by ernestotejeda — APPROVED** [review]

**Pull request merged**